src/avatar/avatar.tsx

```diff
@@ -0,0 +1,90 @@
+import { useCallback, useRef, useState } from "react";
+import { Menu } from "../navbar/menu";
+import { useBlur } from "../util/use-blur";
+import {
+    AvatarBaselineText,
+    AvatarBodySmallText,
+    AvatarButton,
+    Container,
+} from "./avatar.style";
+import { AvatarProps } from "./types";
+
+export const Avatar = <T,>({
+    children,
+    menu,
+    mobile = false,
+    onClick,
+    "data-testid": testId = "avatar",
+    ...otherProps
+}: AvatarProps<T>): JSX.Element => {
+    // =============================================================================
+    // CONST, STATE, REF
+    // =============================================================================
+    const [showMenu, setShowMenu] = useState(false);
+    const ref = useRef<HTMLDivElement>(null);
+
+    // =============================================================================
+    // EVENT HANDLERS
+    // =============================================================================
+    const handleAvatarClick = useCallback(
+        (event: React.MouseEvent<HTMLDivElement>) => {
+            setShowMenu(true);
+            onClick?.(event);
+        },
+        [onClick]
+    );
+
+    const handleBlur = useCallback(() => {
+        setShowMenu(false);
+    }, []);
+
+    // =============================================================================
+    // EFFECTS
+    // =============================================================================
+    useBlur(ref, handleBlur);
+
+    // =============================================================================
+    // RENDER FUNCTIONS
+    // =============================================================================
+    const renderContent = () => {
+        if (!children) {
+            return <></>;
+        }
+
+        if (typeof children === "string") {
+            const TextComponent = mobile
+                ? AvatarBodySmallText
+                : AvatarBaselineText;
+            return (
+                <TextComponent weight="semibold">
+                    {children.charAt(0).toUpperCase()}
```

Comment:
kinda extra but let's do this to handle emoji too

```
String.fromCodePoint(children.codePointAt(0)).toUpperCase()
```

(it still doesn't 100% cover all cases but best effort)